File: 백준/Bronze/33538. Crawling/Crawling.py
import sys
input=lambda:sys.stdin.readline().rstrip()
l=int(input())
n=int(input())
t=float(input())
cnt=0
for _ in range(n):
    ft,bt=map(float,input().split())
    if l/ft+l/bt<t:
        cnt+=1
print("HOPE" if cnt else "DOOMED")

File: 백준/Bronze/33538. Crawling/Crawling.c
#include <stdio.h>
int main(void) {
    int l, n, cnt = 0;
    double t, ft, bt;
    scanf("%d", &l);
    scanf("%d", &n);
    scanf("%lf", &t);
    for (int i = 0; i < n; i++) {
        scanf("%lf %lf", &ft, &bt);
        if (l / ft + l / bt < t)
            cnt += 1;
    }
    printf("%s", cnt > 0 ? "HOPE" : "DOOMED");
    return 0;
}